File: backend/Models/StopTime.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models;

[Table("stop_times")]
public class StopTime
{
    [Column("trip_id")]
    [Required]
    [StringLength(255)]
    public string TripId { get; set; } = null!;

    [Column("arrival_time")]
    [Required]
    public TimeSpan ArrivalTime { get; set; }

    [Column("departure_time")]
    [Required]
    public TimeSpan DepartureTime { get; set; }

    [Column("stop_id")]
    [Required]
    public int StopId { get; set; }

    [Column("stop_sequence")]
    [Required]
    public int StopSequence { get; set; }

    [Column("stop_headsign")]
    [StringLength(255)]
    public string? StopHeadSign { get; set; }

    [Column("pickup_type")]
    [Required]
    public bool PickupType { get; set; }

    [Column("drop_off_type")]
    [Required]
    public bool DropOffType { get; set; }

    [Column("shape_dist_travelled")]
    [Required]
    [Precision(18, 2)]
    public decimal ShapeDistanceTravelled { get; set; }

    [Column("timepoint")]
    [Required]
    public bool Timepoint { get; set; }

    [Column("stop_note")]
    [StringLength(255)]
    public string? StopNote { get; set; }
}

File: backend/TransportRealtime/models/static/stop_time.go
package models

import models "TransportRealtime/models/realtime"

type StaticStopTime struct {
	TripId             string   `json:"tripId"`
	StopId             string   `json:"stopId"`
	StopName           string   `json:"stopName"`
	StopSequence       int      `json:"stopSequence"`
	StopHeadsign       *string  `json:"stopHeadsign"`
	ArrivalTime        int      `json:"arrivalTime"`
	DepartureTime      int      `json:"departureTime"`
	PickupType         int      `json:"pickupType"`
	DropOffType        int      `json:"dropOffType"`
	ShapeDistTravelled *float64 `json:"shapeDistTravelled"`
	Timepoint          *int     `json:"timepoint"`
	StopNote           *string  `json:"stopNote"`
	RouteType          *int     `json:"routeType"`
}

type RealtimeStopTime struct {
	StopId         string           `json:"stopId"`
	StopName       string           `json:"stopName"`
	StopSequence   int              `json:"stopSequence"`
	StopHeadsign   *string          `json:"stopHeadsign"`
	ArrivalTime    *int             `json:"arrivalTime"`
	DepartureTime  *int             `json:"departureTime"`
	ArrivalDelay   *int             `json:"arrivalDelay"`
	DepartureDelay *int             `json:"departureDelay"`
	Status         string           `json:"status"`
	Progress       *string          `json:"progress"`
	Consist        []models.Consist `json:"consist"`
}

type StopRealtimeStopTime struct {
	TripId                 string  `json:"tripId"`
	TripHeadsign           *string `json:"tripHeadsign"`
	ServiceId              string  `json:"serviceId"`
	RouteId                string  `json:"routeId"`
	RouteShortName         *string `json:"routeShortName"`
	RouteType              *int    `json:"routeType"`
	RouteColour            *string `json:"routeColour"`
	StopId                 string  `json:"stopId"`
	StopName               string  `json:"stopName"`
	ArrivalTime            *int    `json:"arrivalTime"`
	ArrivalDelay           *int    `json:"arrivalDelay"`
	DepartureTime          *int    `json:"departureTime"`
	DepartureDelay         *int    `json:"departureDelay"`
	EffectiveArrivalTime   *int    `json:"effectiveArrivalTime"`
	EffectiveDepartureTime *int    `json:"effectiveDepartureTime"`
	PickupType             int     `json:"pickupType"`
	DropOffType            int     `json:"dropOffType"`
	StopType               string  `json:"stopType"`
	IsRealtime             bool    `json:"isRealtime"`
	DisplayTime            int     `json:"displayTime"`
	HasContinuation        bool    `json:"hasContinuation"`
}

type TripRealtimeStopTime struct {
	TripHeadsign           *string          `json:"tripHeadsign"`
	RouteShortName         *string          `json:"routeShortName"`
	RouteType              *int             `json:"routeType"`
	RouteColour            *string          `json:"routeColour"`
	StopId                 string           `json:"stopId"`
	StopName               string           `json:"stopName"`
	ArrivalDelay           *int             `json:"arrivalDelay"`
	DepartureDelay         *int             `json:"departureDelay"`
	EffectiveArrivalTime   *int             `json:"effectiveArrivalTime"`
	EffectiveDepartureTime *int             `json:"effectiveDepartureTime"`
	StopType               string           `json:"stopType"`
	Progress               *string          `json:"progress"`
	IsRealtime             bool             `json:"isRealtime"`
	Consist                []models.Consist `json:"consist"`
	DisplayTime            int              `json:"displayTime"`
	StopProgress           float64          `json:"stopProgress"`
}
